code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/ContentCardSchemaDataUtils.kt

```diff
@@ -0,0 +1,220 @@
+        return AepText(
+            content = DataReader.optString(map, MessagingConstants.ContentCard.UIKeys.CONTENT, null)
+        )
+    }
+
+    /**
+     * Creates an instance of [AepImage] from a map of key-value pairs.
+     *
+     * @param map The map containing the data for [AepImage].
+     * @return An instance of [AepImage].
+     */
+    private fun createAepImage(map: Map<String, Any>): AepImage {
+        return AepImage(
+            url = DataReader.optString(map, MessagingConstants.ContentCard.UIKeys.URL, null),
+            darkUrl = DataReader.optString(map, MessagingConstants.ContentCard.UIKeys.DARK_URL, null)
+        )
+    }
+
+    /**
+     * Creates an instance of [AepButton] from a map of key-value pairs.
+     *
+     * @param map The map containing the data for [AepButton].
+     * @return An instance of [AepButton].
+     */
+    private fun createAepButton(map: Map<String, Any>): AepButton {
+        return AepButton(
+            id = DataReader.optString(map, MessagingConstants.ContentCard.UIKeys.INTERACT_ID, null),
+            actionUrl = DataReader.optString(map, MessagingConstants.ContentCard.UIKeys.ACTION_URL, null),
+            text = DataReader.optTypedMap(Any::class.java, map, MessagingConstants.ContentCard.UIKeys.TEXT, null).let { createAepText(it) }
+        )
+    }
+
+    /**
+     * Creates an instance of [AepDismissButton] from a map of key-value pairs.
+     *
+     * @param map The map containing the data for [AepDismissButton].
+     * @return An instance of [AepDismissButton].
+     */
+    private fun createAepDismissButton(map: Map<String, Any>): AepIcon? {
+        val style = DataReader.optString(map, MessagingConstants.ContentCard.UIKeys.STYLE, MessagingConstants.ContentCard.UIKeys.NONE)
+        return when (style) {
+            MessagingConstants.ContentCard.UIKeys.SIMPLE -> AepIcon(R.drawable.close_filled)
+            MessagingConstants.ContentCard.UIKeys.CIRCLE -> AepIcon(R.drawable.cancel_filled)
+            else -> null
+        }
+    }
+
+    /**
+     * Parses the given [ContentCardSchemaData] to create an [AepUITemplate].
+     *
+     * @param contentCardSchemaData The content card schema data containing the content map.
+     * @return The parsed [AepUITemplate] or null if parsing fails.
+     */
+    internal fun getTemplate(contentCardSchemaData: ContentCardSchemaData): AepUITemplate? {
+        try {
+            val contentMap =
+                contentCardSchemaData.content as? Map<String, Any> ?: throw IllegalArgumentException("Content map is null")
+            val id = contentCardSchemaData.parent.proposition.uniqueId
+
+            val metaAdobeObject =
+                contentCardSchemaData.meta?.get(MessagingConstants.ContentCard.UIKeys.ADOBE)
+
+            metaAdobeObject.let { mapObject ->
+                val templateType =
+                    (mapObject as? Map<String, Any>)?.get(MessagingConstants.ContentCard.UIKeys.TEMPLATE)
```

Comment:
nit: Can we avoid casting by using DataReader?

Reply:
Will do in next PR for UTs